webhooks/virtualmachine/validation/virtualmachine_validator.go

```diff
@@ -153,6 +153,35 @@ func (v validator) ValidateDelete(*pkgctx.WebhookRequestContext) admission.Respo
 	return admission.Allowed("")
 }
 
+// Updates to VM's image are only allowed if it is a Registered VM (used for failover
+// in disaster recovery).
+func (v validator) validateImageOnUpdate(ctx *pkgctx.WebhookRequestContext, vm, oldVM *vmopv1.VirtualMachine) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if f := pkgcfg.FromContext(ctx).Features; f.VMIncrementalRestore && vmopv1util.IsRegisteredVM(*vm) {
```

Comment:
nit: Consider rewriting as follows:

```go
if f := pkgcfg.FromContext(ctx).Features; f.VMIncrementalRestore {
    if vmopv1util.IsRegisteredVM(*vm) {
        // ...
    }
}
```

This makes it easier to remove the feature gate once the FSS is enabled.